File: models/joke.go
// models/joke.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Joke struct {
	ID        uint           `json:"id" gorm:"primaryKey"`
	Content   string         `json:"content"`
	Votes     int            `json:"votes"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}

File: backend/models/Joke.js
// models/Joke.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

// Define the Joke model
const Joke = sequelize.define('Joke', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  content: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  votes: {
    type: DataTypes.INTEGER,
    allowNull: false,
    defaultValue: 0,
  },
}, {
  timestamps: true, // Adds createdAt and updatedAt fields
});

module.exports = Joke;
